lib/schema_plus/core/active_record/schema_dumper.rb

```diff
@@ -68,17 +68,36 @@ def table(table, _)
               env.table.trailer = m[:trailer].split("\n").map(&:strip).reject{|s| s.blank?}
               table_objects = m[:columns].strip.split("\n").map { |col|
                 cs = col.strip
+                result = nil
```

Comment:
At this point I think it's worth splitting up to different functions for each arity. This seems rather convoluted...

Reply:
it's all about priority of matching here..  as I have the expression index case, the generic case, and the column case.

Reply:
once thought I could do, is extract it out into a CONST array of regex/lambda and reduce this block to just iterating those.. Then it'll be a little cleaner? e.g.
```ruby
COLUMN_MATCHERS = [
 [%r{...}, ->(m) {}],
 [%r{...}, ->(m) {}],
].freeze

COLUMN_MATCHERS.find do |r, l| 
    m = cs.match(r)
   l.call(m) unless m.nil?
end
```

Reply:
Yeah that would be awesome, thanks @urkle!

Reply:
@lowjoel take a look now..